### PR TITLE
NOMOS-DATA-AVAILABILITY: New Raw RFC

### PR DESCRIPTION
**Pull request by jimstir**
Open discussion for a Nomos data availability layer raw RFC. Should include all components of the data availability service provided by the Nomos blockchain as well as how they are utilized within the network.

### CODE REVIEW

File: nomos/data-availability-layer.md [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+## Motivation and Goal
+
+Decentralized blockchains require full nodes to verify network transactions by downloading all the data of the network.
+This becomes a problem as the blockchain data grows, full nodes will need more resources to download and 
+store the data while maintaining connection to the network.
+Light nodes on the other hand do not download the entire network data because of it's resource limiting nature.
+This retricts the network from scaling as the network is reliant on full nodes to process transactions.
+and requires light nodes to rely on centralized parties. 
+A blockchain can have light nodes prove the validity of transaction data,
+without the light node to downloading all the blockchain data.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:43]: This is not really the focus. Focus is bandwidth reduction. Splitting data into chunks and knowing who to send what really makes an impact as the network grows.

File: nomos/data-availability-layer.md [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+This allows participants of a zone to access blockchain data in the event that nodes within a zone does not make the data available.
+The service includes data encoding, verification, data availability sampling mechanism, 
+and data retrieval API to solve the data availability problem.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+The data availability service of the Nomos base layer consist of different node roles, dispersal clients, 
+data availibity sampling nodes, and data availability provider nodes.
+All network partipants have a choice in which node role they choose to run.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:55]: They do not. All nodes within the base layer *do* sample and verification. Dispersal is triggered by executor in zones, then by all of the base layer nodes participating in the different DA subnetworks (internal protocol yet to be decided).

File: nomos/data-availability-layer.md [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+
+Nodes MAY decide to provide resources to the data availibilty service of the Nomos base-layer or 
+join a Data Availibility Committee within a zone.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:58]: No such thing as committees here.

File: nomos/data-availability-layer.md [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+MAY selectivly vailidate zk-proofs from the Nomos coordination layer,
+MAY verify data availibility of the base layer for zones they prefer.
+
+Data availibity on the base layer is only a temporary guarantee.
+The data can only be verified for a predetermined time, based on the Nomos network.
+The base layer MUST NOT provide long term incentives or 
+allocate resources to repair missing data.
+Zones act as [Data Availability Committees](#) for their own blockchain state.
+Zones SHOULD provide data availability for their blockchain in the event that light nodes can not access data,
+light nodess MAY utilize the data avilability of the Nomos base layer.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:72]: I think this is incorrect. Or it did change. We do not have committees.

File: nomos/data-availability-layer.md [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+
+### Base-Layer Nodes 
+
+All light node are considered to be Nomos base layer nodes.
+Base-layer nodes MUST NOT process data, 
+but only provide data availability guarantees for a limit amount of time.
+The role of a provider node is to store polynomial commitment schemes for Nomos zones.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:79]: They store the commitments, and a chunk of the original (or extended) data.

File: nomos/data-availability-layer.md [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+Communication between different zones with [Data Availability Committees](#) occur directly.
+Zones use a libp2p swarm to connect to different nodes participating in a zone as a validator,
+.
+It is the responsibility of zones to maintain the swarm.
+When a node in the swarm does not provide access to data,
+light nodes MAY use the Nomos data availbilty layer.
+
+#### Sending Data
+
+Zones are responisble for creating data chunks that need to be stored on the blockchain.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:111]: They are responsible for the encoding (reference to the encoding section maybe?).

File: nomos/data-availability-layer.md [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+The data SHOULD be sent to provider nodes.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:112]: What is a provider node?

File: nomos/data-availability-layer.md [hunk 8]
```diff
@@ -0,0 +1,434 @@
...
+Data chucks are sent with aggregate commitments, a list of row commitments for entire data blob, and 
+a column commitment for the specific data chuck.
+
+##### Dispersal
+
+###### Verification Process
+
+Once encoded, 
+the data is dispersed to different Nomos data availibilty provider nodes that have joined a subnet on the base layer.
+It is RECCOMENDED that the dispersal client sends a column to 4096 provider nodes for better bandwidth optimization.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:202]: They should send a single column to *each* of the 4096 da subnetworks.

File: nomos/data-availability-layer.md [hunk 9]
```diff
@@ -0,0 +1,434 @@
...
+```rs
+// Provider node SHOULD hash using Blake2 algorithm
+// blob_hash : Hash of row_commitment and column_commitiment
+fn sendAsstation () {
+  attestation_hash = hash(blob_hash, DAnode);
+}
+```
+
+The VID certificate is then sent to block builder to be accepted through concensus, 
+as desirbed in [Cryptarchia](#).
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:286]: Gonna remove the VID entirely

File: nomos/data-availability-layer.md [hunk 10]
```diff
@@ -0,0 +1,434 @@
...
+${ \Large verify(C_{agg},C_r, \pi_{C_r})\to true/false}$
+    
+4. If this proof is true, then light nodes wants to opening of the column commitment.
+$node_r$ calculates the $eval$ value and sends it to the light node to be verified.
+
+${ \Large eval(\theta_t,w^{t'-1})\to data_{t'}^{t},\pi_{data_{t'}^{t}} }$
+    
+${ \Large verify(C_t, data_{t'}^t, \pi_{data_{t'}^t})\to true/false }$
+
+If this is true, then this proves that the data chuck has been encoded correctly.
```
Comment (danielSanchezQ) [inline nomos/data-availability-layer.md:321]: All this is WIP as we do not have anything decided yet.

File: nomos/data-availability-layer.md (hunk 1 above)
Comment (jimstir) [reply nomos/data-availability-layer.md:43]: @danielSanchezQ Do you mean not the focus for the network or for the role of a DA node in the BL layer for Nomos? Is focus on bandwidth reduction for light nodes, full nodes or both?

**Comment by jimstir** [comment]
Closing pr because a new raw rfc is being created.

**Pull request closed without merging**